.eslintrc.js

```diff
@@ -48,6 +48,11 @@ module.exports = {
         "no-case-declarations": "warn", // TODO(bkendall): remove, allow to error.
         "no-constant-condition": "warn", // TODO(bkendall): remove, allow to error.
         "no-fallthrough": "warn", // TODO(bkendall): remove, allow to error.
+
+        // Use the TS version of the rule which allows `function foo(this) ...`.
+        // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-invalid-this.md
+        "no-invalid-this": "off",
+        "@typescript-eslint/no-invalid-this": ["error"],
```

Comment:
Can you move these up to the grouping around L34-35? The first block here are "rules that will stay", the second block is "rules that need to be changed/removed". Adding a third block is not necessary.

I'd also suggest formatting as the following (searching for the documentation is fast - pasting the link seems redundant - and keeping the comment on the line keeps the keys sortable):

```js
        "no-invalid-this": "off", // Turned off in favor of @typescript-eslint/no-invalid-this.
        "@typescript-eslint/no-invalid-this": ["error"],
```

Reply:
Done.